projects/igniteui-angular/src/lib/directives/for-of/base.helper.component.ts

```diff
@@ -21,17 +27,34 @@ export class VirtualHelperBaseDirective implements OnDestroy, AfterViewInit {
 
     private _afterViewInit = false;
     private _scrollNativeSize: number;
+    private _observer: ResizeObserver;
+    private _detached = false;
+    protected resizeNotify = new Subject();
+    protected destroy$ = new Subject<any>();
+
 
     ngAfterViewInit() {
         this._afterViewInit = true;
+        const destructor = takeUntil<any>(this.destroy$);
+        this.resizeNotify.pipe(destructor,
+        debounceTime(40, undefined))
+        .subscribe((event) => {
+            this._zone.runTask(() => {
+                this.handleMutations(event);
+            });
+        });
     }
 
     @HostListener('scroll', ['$event'])
     onScroll(event) {
         this.scrollAmount = event.target.scrollTop || event.target.scrollLeft;
     }
-    constructor(public elementRef: ElementRef, public cdr: ChangeDetectorRef) {
+    constructor(public elementRef: ElementRef, public cdr: ChangeDetectorRef, protected _zone: NgZone, @Inject(DOCUMENT) public document) {
         this._scrollNativeSize = this.calculateScrollNativeSize();
+        this._zone.runOutsideAngular(() => {
+            this._observer = new ResizeObserver((event) => this.resizeNotify.next(event));
+            this._observer.observe(this.nativeElement);
```

Comment:
We have a helper function to return an observable from DOM observers.
```typescript
import { resizeObservable } from '../../core/utils';

resizeObservable(this.nativeElement).pipe(destructor)
                .subscribe((event) => this._zone.runTask(() => this.handleMutations(event)));
```
You won't need the `resizeNotify` subject and the manual disconnect of the observer on destroy.